pkg/backingstore/reconciler.go

```diff
@@ -928,6 +928,16 @@ func (r *Reconciler) ReconcilePool() error {
 }
 
 func (r *Reconciler) reconcilePvPool() error {
+	if r.Secret.StringData["AGENT_CONFIG"] == "" {
+		res, err := r.NBClient.GetHostsPoolAgentConfigAPI(nb.GetHostsPoolAgentConfigParams{
+			Name: r.BackingStore.Name,
+		})
+		if err != nil {
```

Comment:
not sure it's necessary here, but make sure you don't need to handle different errors in a different way (persistent vs. temp error)

Reply:
well I'm currently unaware of any persistent error here. you think about anything specific? thinking about rpc failures that should be temporal - not sure about others... especially when the pool suppose to exist if we got here and it won't be any other type of pool other than hosts